File: Week1/채우석/Boj10828.java
package Week1.채우석;


import java.io.*;
import java.util.Stack;
import java.util.StringTokenizer;

// 스택
public class Boj10828 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        int N = Integer.parseInt(br.readLine());

        Stack<Integer> stack = new Stack<>();

        for (int i = 0; i < N; i++) {
            StringTokenizer st  = new StringTokenizer(br.readLine());
            String command = st.nextToken();

            switch (command) {
                case "push":
                    stack.add(Integer.parseInt(st.nextToken()));
                    break;
                case "pop":
                    bw.write(stack.isEmpty() ? "-1\n" : stack.pop() + "\n");
                    break;
                case "size":
                    bw.write(stack.size() + "\n");
                    break;
                case "empty":
                    bw.write(stack.isEmpty() ? "1\n" : "0\n");
                    break;
                case "top":
                    bw.write(stack.isEmpty() ? "-1\n" : stack.peek() + "\n");
                    break;
            }
        }
        bw.flush();
        br.close();
        bw.close();
    }
}

File: Week1/김민수/BOJ_10828.py
#스택

import sys

n = int(sys.stdin.readline())
stack =[]

for i in range(n):
    c = sys.stdin.readline().split()

    if c[0]=='push':
        stack.append(c[1])

    elif c[0]=='pop':
        if len(stack)==0:
            print(-1)
        else:
            print(stack.pop())

    elif c[0] == 'size':
        print(len(stack))

    elif c[0] == 'empty':
        if len(stack)==0:
            print(1)
        else:
            print(0)

    elif c[0] == 'top':
        if len(stack) == 0:
            print(-1)
        else:
            print(stack[-1])